Submit pr guide en (#500)
PaddlePaddle/FluidDoc: local_dev_guide_en.md

# Guide of local development

You will learn how to develop programs in local under the guide of this document.

## Requirements of code
- Please refer to the coding note format of [Doxygen](http://www.stack.nl/~dimitri/doxygen/) 
- Make sure that option of builder `WITH_STYLE_CHECK` is on and the build could pass through the code style check.
- Unit test is a need for all codes.
- Pass through all unit tests.
- Please follow [regulations of submitting code](#regulations of submitting code).

It tells you how to submit code as follows.
## [Fork](https://help.github.com/articles/fork-a-repo/)

Transfer to the homepage of Github [PaddlePaddle](https://github.com/PaddlePaddle/Paddle),and then click button`Fork` to generate the repository under your own file directory,such as <https://github.com/USERNAME/Paddle>.

## Clone

Clone remote repository to local:

```bash
➜  git clone https://github.com/USERNAME/Paddle
➜  cd Paddle
```

## Create local branch

> At present [Git stream branch model](http://nvie.com/posts/a-successful-git-branching-model/) is applied to Paddle to undergo task of development,test,release and maintenance.Please refer to [branch regulation of Paddle](https://github.com/PaddlePaddle/Paddle/blob/develop/doc/design/releasing_process.md#paddle-分支规范) about details。

Reviewer: 英文标点后请空一格

> All development tasks of feature and bug fix should be finished in a new branch which is extended from `develop` branch.

Reviewer: ```suggestion
All feature and bug fix development work should be done on a new branch usually created from `develop` branch.
```

> Create and change into a new branch with command `git checkout -b`.

Reviewer: ```suggestion
Create and switch to a new branch with command `git checkout -b`.
```

```bash
➜  git checkout -b my-cool-stuff
```

> Atentions should be paid that current branch be clean before checkout,otherwise the untracked files will also be shifted to the new branch,which could be viewed with command `git status`.

Reviewer: ```suggestion
It is worth noting that before the checkout, you need to keep the current branch directory clean, otherwise the untracked file will be brought to the new branch, which can be viewed by git status. `git status`.
```

## Use `pre-commit` hook

> Paddle developers manage Git pre-commit hook with [pre-commit](http://pre-commit.com/),which could help us to fomulate source code(C++，Python) and automatically review certain details,such as only one EOL of every file and no large file added to Git,before your commit.

Reviewer: ```suggestion
Paddle developers use the [pre-commit](http://pre-commit.com/) tool to manage Git pre-commit hooks. It helps us format the source code (C++, Python) and automatically check some basic things before committing (such as having only one EOL per file, not adding large files in Git, etc.).
```

> `pre-commit` test is part of unit tests of Travis-CI.PR that doesn't meet the requirement of hook can't be commit to Paddle.Install `pre-commit` first and then run it on current directory：

Reviewer: ```suggestion
The `pre-commit` test is part of the unit test in Travis-CI. A PR that does not satisfy the hook cannot be submitted to Paddle. Install `pre-commit` first and then run it in current directory：
```

```bash
➜  pip install pre-commit
➜  pre-commit install
```

> You can modify the format of C/C++ source code with `clang-format` in Paddle.Make sure the version of `clang-format` is above 3.8.

Reviewer: ```suggestion
Paddle uses `clang-format` to adjust the C/C++ source code format. Make sure the version of `clang-format` is above 3.8.
```

> Note：There are differences between the installation of `yapf` with `pip install pre-commit` and that with `conda install -c conda-forge pre-commit`.Paddle developers use `pip install pre-commit`.

Reviewer: ```suggestion
Note: The `yapf` installed by `pip install pre-commit` and `conda install -c conda-forge pre-commit` is slightly different. Paddle developers use pip install pre-commit.
```

> ## Starting the development

Reviewer: ```suggestion
## Start development
```

> I delete one line of README.md and create a new file at the example.

Reviewer: ```suggestion
In this example, one line in README.md is deleted and a new file is created.
```

> Check current status with `git status`,followed by the appearance of changes under current directory.Meanwhile,you can also check the specific modifications of the file with `git diff`.

Reviewer: ```suggestion
View the current state via `git status` , which will prompt some changes to the current directory, and you can also view the file's specific changes via `git diff` .
```

```bash
➜  git status
On branch test
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	modified:   README.md

Untracked files:
  (use "git add <file>..." to include in what will be committed)

	test

no changes added to commit (use "git add" and/or "git commit -a")
```

> ## Constructure and test

Reviewer: ```suggestion
## Build and test
```
